iroh-bytes/src/provider.rs

```diff
@@ -359,23 +364,22 @@ pub async fn read_request(mut reader: quinn::RecvStream, buffer: &mut BytesMut)
 /// close the writer, and return with `Ok(SentStatus::NotFound)`.
 ///
 /// If the transfer does _not_ end in error, the buffer will be empty and the writer is gracefully closed.
-pub async fn transfer_collection<E: EventSender>(
+pub async fn transfer_collection<D: BaoCollection, E: EventSender>(
```

Comment:
Hmm, I wonder if `BaoCollection` is a good name for this trait. Kinda hard to understand that that is the database.

Reply:
It is "A generic collection of blobs with precomputed outboards" - only the read part. Also not super happy with the name, especially because it collides with the other `Collection`. But Database is also not a good name since that encompasses both the read and write part and will contain more stuff.

BlobMap? BaoMap?

Reply:
I like `BaoMap` it differentiates from `Collection` and makes it clear anything "bao" like is stored it.